<launch>
  <group name='bax'>
     <node pkg="baxter_lcm" name ="repeater_r" type = "LCMToRos.py" output="screen">
        <param name="id"  value="100" />
        <param name="lr"  value="R"/>
    </node>

     <node pkg="baxter_lcm" name ="listener_r" type = "BaxterInterface.py" output="screen">
        <param name="part" value="right" />
        <param name="id"  value="100" />
     </node>
     <node pkg="baxter_lcm" name ="listener_r_g" type = "BaxterInterface.py" output="screen">
        <param name="part" value="right_gripper" />
        <param name="id"  value="100" />
     </node>

     <node pkg="baxter_lcm" name ="listener_r_r" type = "BaxterInterface.py" output="screen">
        <param name="part" value="right_range" />
        <param name="id"  value="100" />
     </node>

     <node pkg="baxter_lcm" name ="listener_m_cam" type = "BaxterInterface.py" output="screen">
        <param name="part" value="monitor_camera_r" />
        <param name="id"  value="100" />
     </node>

     <node pkg="baxter_lcm" name ="listener_r_cam" type = "BaxterInterface.py" output="screen">
        <param name="part" value="right_camera" />
        <param name="id"  value="100" />
     </node>
     
     <node pkg="baxter_lcm" name ="self_sender_r" type = "LCMSelf.py" output="screen">
        <param name="id"    value="100" />
        <param name="type"    value="Baxter" />
        <param name="ability"   value="idk" />
        <param name="channels"  value="right_lcm_channel|right_lcm_valid|right_lcm|right_lcm_cmd|right_trigger_lcm|right_lcm_currentpos|right_lcm_range|confidence_threshold_lcm|right_lcm_camera"/>
        <param name="queryChannel"     value="control_query|100" />
        <param name="leftNDIChannel"   value="NDIBOX (Logitech Webcam C930e 1)" />
        <param name="rightNDIChannel"  value="NDIBOX (Logitech Webcam C930e 0)"/>
        <param name="broadcastChannel" value="global_self"/>
     </node>

     <node pkg="baxter_lcm" name ="QueryRos_r" type = "QueryToRos.py" output="screen">
        <param name="queryChannel"  value="control_query|100" />
     </node>

     <node pkg="baxter_lcm" name ="ConfidenceThresholdRos_r" type = "ConfidenceThresholdRos.py" output="screen">
        <param name="confidenceThresholdChannel"  value="confidence_threshold_lcm-100" />
     </node>

     <node pkg="baxter_lcm" name ="info_sender_r" type = "LCMInfo.py" output="screen">
        <param name="id"  value="100" />
        <param name="lcmChannel"  value="global_info"/>
        <param name="lr"  value="R"/>
     </node>
    
  </group>
</launch>
